dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -877,6 +906,160 @@ public void download(final List<String> args) {
         }
     }
 
+
+    /**
+     * Creates a WES API object and sets the endpoint.
+     *
+     * @param wesUrl URL to WES endpoint
+     */
+    public WorkflowExecutionServiceApi getWorkflowExecutionServiceApi(String wesUrl) {
+        WorkflowExecutionServiceApi clientWorkflowExecutionServiceApi = new WorkflowExecutionServiceApi();
+        ApiClient wesApiClient = clientWorkflowExecutionServiceApi.getApiClient();
+
+        INIConfiguration config = Utilities.parseConfig(this.getConfigFile());
+        SubnodeConfiguration configSubNode = null;
+        try {
+            configSubNode = config.getSection("WES");
+        } catch (Exception e) {
+            //?????what should the return code be?????
+            exceptionMessage(e, "Could not get WES section", ENTRY_NOT_FOUND);
+        }
+
+        if (wesUrl == null || wesUrl.isEmpty()) {
+            String wesEndpointUrl = configSubNode.getString("url");
+            if (wesEndpointUrl == null || wesEndpointUrl.isEmpty()) {
+                errorMessage("No WES URL found in config file and no WES URL entered on command line.", CLIENT_ERROR);
+            }
+            out("WES endpoint url is: " + wesEndpointUrl);
+            wesApiClient.setBasePath(wesEndpointUrl);
+        } else {
+            wesApiClient.setBasePath(wesUrl);
+        }
+
+        /**
+         * Setup authentication credentials for the WES URL
+         */
+        //TODO: somehow make this case insensitive?
+        String wesAuthorizationTypeCredentials = configSubNode.getString("authorization", null);
+        String wesAuthorizationType = null;
+        String wesAuthorizationCredentials = null;
+        //out("wes config authorization string is " + wesAuthorizationTypeCredentials);
+        if (wesAuthorizationTypeCredentials != null) {
+            String[] wesAuthorizationTypeCredentialsArray = wesAuthorizationTypeCredentials.split("\\s+");
```

Comment:
I think you should pass in the credentials "as is", if they are present, and not do all these tests.

IOW, if your config file contains (I don't know the exact syntax):
```
[wes]
authorization: Bearer sometoken
```

Just set the authorization header to value to `Bearer sometoken`. Right now you split the string to see if the value starts with `Bearer` or `Basic`, and fail if it is not set to either of those. 

There are other schemes besides Bearer and Basic (not that I've ever used them :) ), so you'd have to checks for all of those as well. Why bother, IMHO -- just blindly set the authorization header.